Gems/ROS2/Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -0,0 +1,248 @@
+        {
+            // if a ticket for this spawnable was not created but the spawnable name is correct, create the ticket and then use it to
+            // spawn an entity
+            auto spawnable = m_spawnables.find(spawnable_name);
+            m_tickets.emplace(spawnable->first, AzFramework::EntitySpawnTicket(spawnable->second));
+        }
+
+        auto spawner = AZ::Interface<AzFramework::SpawnableEntitiesDefinition>::Get();
+
+        AzFramework::SpawnAllEntitiesOptionalArgs optionalArgs;
+
+        AZ::Transform transform;
+
+        if (spawn_points.contains(spawn_point_name))
+        {
+            transform = spawn_points.at(spawn_point_name).pose;
+        }
+        else
+        {
+            transform = { AZ::Vector3(request->initial_pose.position.x, request->initial_pose.position.y, request->initial_pose.position.z),
+                          AZ::Quaternion(
+                              request->initial_pose.orientation.x,
+                              request->initial_pose.orientation.y,
+                              request->initial_pose.orientation.z,
+                              request->initial_pose.orientation.w),
+                          1.0f };
+        }
+
+        optionalArgs.m_preInsertionCallback = [this, transform, spawnable_name](auto id, auto view)
+        {
+            PreSpawn(id, view, transform, spawnable_name);
+        };
+
+        spawner->SpawnAllEntities(m_tickets.at(spawnable_name), optionalArgs);
+
+        response->success = true;
+    }
+
+    void ROS2SpawnerComponent::PreSpawn(
+        AzFramework::EntitySpawnTicket::Id id [[maybe_unused]],
+        AzFramework::SpawnableEntityContainerView view,
+        const AZ::Transform& transform,
+        const AZStd::string& spawnable_name)
+    {
+        if (view.empty())
+        {
+            return;
+        }
+        AZ::Entity* root = *view.begin();
+
+        // TODO: probably it would be better to use TransformBus here
+        auto* transformInterface = root->FindComponent<AzFramework::TransformComponent>();
+        transformInterface->SetWorldTM(transform);
+
+        AZStd::string instanceName = AZStd::string::format("%s_%d", spawnable_name.c_str(), m_counter++);
+        for (AZ::Entity* entity : view)
+        { // Update name for the first entity with ROS2Frame in hierarchy (left to right)
+            const auto* frameComponent = Utils::GetGameOrEditorComponent<ROS2FrameComponent>(entity);
+            if (frameComponent)
+            {
+                entity->SetName(instanceName);
+                break;
+            }
+        }
+    }
+
+    const AZ::Transform& ROS2SpawnerComponent::GetDefaultSpawnPose() const
+    {
+        return m_defaultSpawnPose;
+    }
+
+    void ROS2SpawnerComponent::GetSpawnPointsNames(
+        const ROS2::GetSpawnPointsNamesRequest request, ROS2::GetSpawnPointsNamesResponse response)
+    {
+        for (auto spawn_point : GetSpawnPoints())
```

Comment:
This spawn_point variable should be lowerCamelCase.
```suggestion
        for (auto spawnPoint : GetSpawnPoints())
```